### ISSUE TITLE
Suggestion: More varied parts

### ISSUE THREAD

**Issue by pyalot** [ChrisAdderley/HeatControl#4]
There is one other mod that dabbles with heat control components (https://kerbalstuff.com/mod/723/Heat%20Management) and they offer basically 3 classes of inline parts:
- Heatsinks: Those are parts which have a high thermal mass (both internal and on the skin) and quickly drain heat from connected components or their own skin to their internal heat. They only reluctantly release heat again to attached components and they're not good at radiating heat.
- Thermal washers: These parts neither take on heat easily nor do they pass it on easily, they're basically thermal insulation.
- Active heat management system: This part works much like an ablator in that it consumes NH3 to get rid of excess heat. Unlike an ablator however it can drain heat from any part of the ship and if supplied with (copious) amounts of electricity it can "regenerate" the NH3 supply.

I think that's a cool combination of parts that're more varied than just radiators.

I'd suggest as food for further thought the following parts for heat control:
- Inline and attachable heatsinks (just like heat management ones)
- Heatsink nosecones and leading edges (basically aerodynamic heatsinks)
- Inline and attachable insulators in various useful shapes
- Inline and attachable active heat management systems, maybe using a similar system to Heat Management, or perhaps just being "from anywhere on the ship heat attractors" and then need to be fitted with additional cooling capacity of some kind
- Evaporators: Like nuclear cooling towers or cooling boosters in some WW2 aircraft, where a liquid is boiled off a suitable structure to cool it down. You'd have to pack tanks of evaporator liquid and perhaps you'd need to pick the right one (depending on ambient temperature/pressure you intend to use it in)
- Waste heat generators: more or less self explanatory, but take for instance a sterling engine, that's a good example of how you generate power from heat (without a consumable working fluid). You just need a temperature differential. They're also heat exchangers of course.
- Heaters: Consumes electricity, makes heat, would be useful for testing and sometimes to boot components that need a level of heat to operate efficienty.
- Coolers: Distinctly different from radiators, these components are also heat exchange systems, but much like say a fridge, they go cold. It could be useful to use a cooler in combination with a heatsink for instance if you know you're going to experience rapid heating soon, you could supercool a part with a large thermal mass that could then act as a more efficient waste heat capture device (while it warms up)

I'm sure some of these ideas make little sense, and there might be many ideas I missed, just some thoughts.

**Comment by ChrisAdderley** [comment]
Yes some of these are probably planned. Heat insulators (aka thermal washers), for example, were in the previous versions (0.1ish). I removed them for 2 reasons - one that 1.05's active radiators don't really care where a part is. If you have radiators on your ship, an insulator isn't as useful. The second is that the models weren't that good looking. They might come back with some new models in the future.

Heatsinks I feel have limited utility at the moment. I have mixed opinions of this, but I just don't see the point in investing model effort there right now.  

Re: open cycle systems. I did want to do a few types at some point, including ones that worked on local atmosphere, ones that worked on local liquids, and ones that worked on evaporative principles. So that way you would be able to represent the full set of nuclear cooling options, radiators, evaporative cooling towers and heat exchange with local water. It would make it a lot easier to build surface reactors and airborne reactors than relying on fragile radiators. 

Waste heat generators - check for a seebek generator mod. Someone made this and I have no plans to do that.

Things like coolers and heaters I would also not hope for - they would need a purpose vs a theoretical purpose. As far as I'm aware there aren't any stock or mod parts that have such specific temperature ranges to act on.

**Comment by pyalot** [comment]
>  If you have radiators on your ship, an insulator isn't as useful.

An insulator can be quite useful if you have a (potentially) hot part, you'd like to insulate from a part that's got a lower heat tolerance. For instance, a rockets nosecone might get quite hot due to convective heating, and that heat will percolate trough any attached part, regardless if you have a radiator or not. Now if one of the attached parts happens to be say, a lander can, it can be problematic because of the large difference in heat tolerance of a nosecone and the lander can.

> Heatsinks I feel have limited utility at the moment. I have mixed opinions of this, but I just don't see the point in investing model effort there right now.

I've experimented quite a bit with the heatsinks from the heat management mod, and after not understanding their use at all, I came around to appreciate them quite a bit. So one of the things they're quite handy for is to attach them to parts that might get hot, but for which you do not have immediately enough cooling capacity. They can act as a buffer to soak up some heat till you can get rid of it. If your mission profile does not include continuously high heat production but occasional hotspots (like re-entry) that can be quite useful. Another use that's quite odd is basically a design like that:

![2015-12-27_00001](https://cloud.githubusercontent.com/assets/592826/12012189/16b99f62-acea-11e5-80dd-7de6ec10425a.jpg)

Here I'm using a heatsink less for its capacity to soak up heat (though that certainly helps) and more for its capacity to quickly transfer heat from the skin to internal. Once internal, the large cooling capacity of the AHMS comes to bear on it. This allows you to fly at high speeds trough atmosphere for extended periods of time (albeit at the edge of overheat of everything).

In a similar fashion, (altough much more extreme) you can use the heatsink as a heat shield from radiation. If you stack the biggest (by mass and surface area) heatsink behind a whole tower of AHMS'es and top it off with another tower of nuclear power plants, it's actually feasible to go into a 600km above the sun orbit (because the heatsink is so good in soaking heat into internal from skin).
